src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -77,10 +79,20 @@ export function QuestionnaireForm({
     prefetch: !!questionnaireSlug && !FIXED_QUESTIONNAIRES[questionnaireSlug],
   });
 
-  const { mutate: submitBatch, isProcessing } = useMutation(
-    routes.batchRequest,
-    { silent: true },
-  );
+  const { mutate: submitBatch, isPending } = useMutation({
+    mutationFn: mutate(routes.batchRequest, { silent: true }),
+    onSuccess: () => {
+      toast.success(t("questionnaire_submitted_successfully"));
+      onSubmit?.();
+    },
+    onError: (error: HTTPError) => {
```

Comment:
it's already type safe right? we have it the type for error declared in the tanstack's type declarations globally.

```suggestion
    onError: (error) => {
```

Reply:
yaaap